CMakeLists.txt

```
@@ -56,78 +56,129 @@ if( ENABLE_TESTING )
     find_package( GTest REQUIRED )
 
     set ( TEST_DRIVE_FIND_METHOD fetch )
-    # option TEST_DRIVE_BUILD_TESTING is hard-coded to ON
+    # Option TEST_DRIVE_BUILD_TESTING is hard-coded to ON, could be spared
     find_package( test-drive REQUIRED )
 endif()
 
+# Optionally use PackageCompiler.jl to build standalone libtrixi.so
+option( USE_PACKAGE_COMPILER "Build standalone libtrixi.so using PackageCompiler.jl" )
 
+# Fortran mod file location
+set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR})
 
-# Library target
-add_library ( ${PROJECT_NAME} SHARED
-    src/api.c
-    src/api.f90
-    src/auxiliary.h
-    src/auxiliary.c
-    src/trixi.h
-)
+if( USE_PACKAGE_COMPILER )
+    if ( NOT DEFINED JULIA_PROJECT_PATH )
+        message( FATAL_ERROR "JULIA_PROJECT_PATH needs to be set for PackageCompiler.jl.")
+    endif()
+    if ( ENABLE_TESTING )
+        message( NOTICE "Testing is not supported when PackageCompiler is used.")
+    endif()
 
-# Include directories, private
-target_include_directories ( ${PROJECT_NAME} PRIVATE src )
+    # Copy source with initialization routines; an object file will be created
+    file( MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/build-pc )
+    file( COPY ${CMAKE_SOURCE_DIR}/LibTrixi.jl/lib/init.c
+          DESTINATION build-pc/ )
```

I think this is not the "proper" way to do it, since it will only ever run once at configure time but not at build time. That is, the `init.c` file will be copied when CMake itself is run, but not whenever we run `make`. Thus, any changes to `init.c` *after* calling CMake will not be picked up.

I think the proper way to do this is by using a custom command to create the file, since then the output can be added as a dependency (see also [this blog post](https://crascit.com/2017/04/18/generated-sources-in-cmake-builds/)).

If I am not mistaken, something like this should work:
```suggestion
    add_custom_command( OUTPUT ${CMAKE_BINARY_DIR}/build-pc/init.c
                        COMMENT "Copying `init.c` to build folder..."
                        COMMAND ${JULIA_EXECUTABLE} -e 'cp(ARGS[1], ARGS[2])'
                                ${CMAKE_SOURCE_DIR}/LibTrixi.jl/lib/init.c
                                ${CMAKE_BINARY_DIR}/build-pc/init.c )
```

---

Btw, I used the Julia command to be more platform agnostic. You could of course just replace that by a `cp`...

---

Good point!

With [75344f5](https://github.com/trixi-framework/libtrixi/pull/128/commits/75344f58cb3e5ca694b0bc636a1f2f650dd6f00c) it seems to work:

- building from scratch works (as before)
- simply calling make again does not do anything (as before)
- in case `init.c` is missing in the build directory or modified in the source directory everything is rebuild

---

Great, thanks for checking it out!